src/amperity/gocd/secret/vault/plugin.clj

```diff
@@ -110,25 +128,83 @@
 ;; configuring a secret backend in GoCD.
 (defmethod handle-request "go.cd.secrets.get-metadata"
   [_ _ _]
-  ;; TODO: how does the plugin authenticate?
   {:response-code    200
    :response-headers {}
-   :response-body    [{:key      :vault_addr
-                       :metadata {:required true :secure false}}]})
+   :response-body    (mapv (fn [input-key]
+                             {:key      input-key
+                              :metadata (-> input-schema input-key :metadata)})
+                           (keys input-schema))})
+
+
+(defn- input-error-message
+  "If the input field is valid, does nothing, if it's not, returns a string describing the error.
+
+  Params:
+  - `field-key`: the input field you are validating
+  - `field-value`: the inputted value which you wish to verify
+  - `field-label`: the human readable representation of the field key"
+  [field-key field-value]
+  (let [field-model (field-key input-schema)
+        validate-func-errors (remove #((:func %)  field-value)
+                                     (:validate-funcs field-model))]
+    (cond
+      ;; field is required but empty
+      (and (-> field-model :metadata :required) (str/blank? field-value))
+      (str (:label field-model) " is required")
+
+      ;; field is not empty and there is an error
+      (not (or (str/blank? field-value) (empty? validate-func-errors)))
+      (:message (first validate-func-errors))
```

Comment:
since the user is getting back just the first validation error, the thing finding errors could just stop at the first one:

```clojure
validation-error (some #(% field-value) (:validate-funcs field-model))
```

since `some` returns the first truthy predicate result from a collection

```clojure
user=> (some
 #(% "a")
 [(fn [x] (when (= "b" x) "found a b"))
  (fn [x] (when (= 'a x) "found a 'a"))
  (fn [x] (when (= "a" x) "found an a"))])
"found an a"
```

Reply:
`remove` returns a lazy seq, so I think it would stop at the first too? But `some` is simpler/more readable anyway so either way I'm making the change 👍